### DISCUSSION TITLE
Can't connect to codespaces from VS Code Desktop: Auth failure

### DISCUSSION THREAD

**Discussion by melink14** [community/community#18905] [Codespaces] [answered]
Hi,

I recently started to use Codespaces and it works great using the browser version of Visual Studio Code!

However, if I try to open the same codespaces from my desktop version of Visual Studeio (1.68.1) then the connection fails with an auth error:
```
[Client V] Initializing authentication
[Authentication W] Failed to load session: undefined
[Authentication W] Failed to load session: undefined
[API I] Signing out due to unsuccessful sign in attempt
```

After this initial auth failure the remote connection thinks that I'm not logged in and thus fails repeatedly until it gives up and hangs until closed with the error message "Please sign in to the GitHub Codespaces extension before connecting to a codespace."

The rest of my github auth based extensions work so I'm not quite sure what to do. So far I've done:
- Re-installed codespaces extension
- Logged out of github in VS code.
- Tried to create new codespace and connect from inside VS Code.

Thanks!

**Comment by melvincwng** [comment]
Hi @melink14, try submitting a ticket on GitHub Support here: https://support.github.com/

They usually reply pretty quickly (within a few hours to 1-2 days) for support-related issues.

Hope this helps you for the time being.

**Reply by melink14** [reply]
Thanks melvincwng! I just assumed that community support was the only option but I've submitted a ticket directly now.

Thanks!

**Reply by melvincwng** [reply]
@melink14, no problem, glad that it helped you!

If you find the answer useful, kindly help to 'Mark as answer' so that others can know this issue is answered.

Thanks!

**Comment by melink14** [comment, marked as answer]
After working with support a bit we found that it was related to one of my settings which was incapatibile with Codespaces. In order to work around an old bug in a different Github auth supported extension I had the following lines in my settings.json:
```
    "remote.extensionKind": {
      "vscode.github-authentication": ["workspace"]
    },
```

This tries to install Github Auth on the remote which breaks auth in Codespaces it seems. Removing this setting makes Codespaces work again!

In general, it seems like a good debugging step for this type of problem would be binary searching through settings and extensions to see if any might cause a problem.
